### ISSUE TITLE
bug: EXDATE does not exclude modified event

### ISSUE THREAD

**Issue by zoifar** [niccokunzmann/python-recurring-ical-events#163]
EXDATE does not exclude a modified instance for an event with higher SEQUENCE and the same UID.

```
import recurring_ical_events
import icalendar
ical_string="""BEGIN:VCALENDAR
BEGIN:VEVENT
DTSTAMP:20240820T210909Z
DTSTART;VALUE=DATE:20240729
DTEND;VALUE=DATE:20240805
SUMMARY:test123
CATEGORIES:other
UID:111
ORGANIZER:aaa
RRULE:FREQ=WEEKLY;INTERVAL=3;BYDAY=MO
CREATED:20240311T051101Z
LAST-MODIFIED:20240311T051101Z
SEQUENCE:1
END:VEVENT
BEGIN:VEVENT
DTSTAMP:20240820T210909Z
DTSTART;VALUE=DATE:20240819
DTEND;VALUE=DATE:20240822
SUMMARY:test123
CATEGORIES:other
UID:111
ORGANIZER:aaa
RRULE:FREQ=WEEKLY;INTERVAL=3;BYDAY=MO
RECURRENCE-ID;VALUE=DATE:20240819
CREATED:20240729T132247Z
LAST-MODIFIED:20240729T132247Z
SEQUENCE:1
END:VEVENT
BEGIN:VEVENT
DTSTAMP:20240820T210908Z
DTSTART;VALUE=DATE:20240729
DTEND;VALUE=DATE:20240805
SUMMARY:test123
CATEGORIES:other
UID:111
ORGANIZER:aaa
RRULE:FREQ=WEEKLY;INTERVAL=3;BYDAY=MO
EXDATE;VALUE=DATE:20240819
CREATED:20240311T051101Z
LAST-MODIFIED:20240729T133342Z
EXDATE;VALUE=DATE:20240819
SEQUENCE:2
END:VEVENT
END:VCALENDAR
""" 


calendar = icalendar.Calendar.from_ical(ical_string)
events=recurring_ical_events.of(calendar).at([2024,8])

for event in events:
    start = event["DTSTART"].dt
    duration = event["DTEND"].dt - event["DTSTART"].dt
    print("start {} duration {}".format(start, duration))
```

**ICS file**

```
BEGIN:VCALENDAR
BEGIN:VEVENT
DTSTAMP:20240820T210909Z
DTSTART;VALUE=DATE:20240729
DTEND;VALUE=DATE:20240805
SUMMARY:test123
CATEGORIES:other
UID:111
ORGANIZER:aaa
RRULE:FREQ=WEEKLY;INTERVAL=3;BYDAY=MO
CREATED:20240311T051101Z
LAST-MODIFIED:20240311T051101Z
SEQUENCE:1
END:VEVENT
BEGIN:VEVENT
DTSTAMP:20240820T210909Z
DTSTART;VALUE=DATE:20240819
DTEND;VALUE=DATE:20240822
SUMMARY:test123
CATEGORIES:other
UID:111
ORGANIZER:aaa
RRULE:FREQ=WEEKLY;INTERVAL=3;BYDAY=MO
RECURRENCE-ID;VALUE=DATE:20240819
CREATED:20240729T132247Z
LAST-MODIFIED:20240729T132247Z
SEQUENCE:1
END:VEVENT
BEGIN:VEVENT
DTSTAMP:20240820T210908Z
DTSTART;VALUE=DATE:20240729
DTEND;VALUE=DATE:20240805
SUMMARY:test123
CATEGORIES:other
UID:111
ORGANIZER:aaa
RRULE:FREQ=WEEKLY;INTERVAL=3;BYDAY=MO
EXDATE;VALUE=DATE:20240819
CREATED:20240311T051101Z
LAST-MODIFIED:20240729T133342Z
EXDATE;VALUE=DATE:20240819
SEQUENCE:2
END:VEVENT
END:VCALENDAR
```

**Expected behavior**

only non EXDATE events returned
`start 2024-07-29 duration 7 days, 0:00:00
`

**Console output**

```
start 2024-07-29 duration 7 days, 0:00:00
start 2024-08-19 duration 3 days, 0:00:00
```

**Version:**

3.1.0

```shell
pip list                         
Package               Version
--------------------- -----------
icalendar             5.0.13
pip                   23.3.1
python-dateutil       2.9.0.post0
pytz                  2024.1
recurring-ical-events 3.1.0
setuptools            69.0.2
six                   1.16.0
tzdata                2024.1
x-wr-timezone         0.0.7
```

**Additional context**

This is generated by atlassian confluence calendar when an instance of an event is modified and deleted.

<small>
<hr/>
We're using <a href="https://polar.sh/niccokunzmann">Polar.sh</a> so you can upvote and help fund this issue. We receive the funding once the issue is completed & confirmed by you. Thank you in advance for helping prioritize & fund our work.
</small>

<a href="https://polar.sh/niccokunzmann/python-recurring-ical-events/issues/163">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/niccokunzmann/python-recurring-ical-events/issues/163/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/niccokunzmann/python-recurring-ical-events/issues/163/pledge.svg">
</picture>
</a>

**Comment by niccokunzmann** [comment]
Thanks for the report!

**Comment by niccokunzmann** [comment]
@zoifar We had #148 before. I identified two edge cases for which I tested to my best understanding:

- https://github.com/niccokunzmann/python-recurring-ical-events/blob/main/test/calendars/issue_148_edge_case_1.ics
- https://github.com/niccokunzmann/python-recurring-ical-events/blob/main/test/calendars/issue_148_edge_case_2.ics

Could you check with the atlassian confluence calendar what happens at these dates?

    date(2024, 7, 2)
    date(2024, 7, 1)
    date(2024, 7, 29)

I wonder what happens in other calendar implementations at these dates...

## Outlook

edge case 1:
![image](https://github.com/user-attachments/assets/b7f7e274-2043-4a45-b5d2-178858150e94)

edge case 2:
![image](https://github.com/user-attachments/assets/07da6d99-fcb6-4c06-90ec-7e37b40cb5ee)

## Thunderbird

![image](https://github.com/user-attachments/assets/fda2e0f9-5970-429b-85ec-9830aca18b7e)

Edge case 1 and 2 look the same:

![image](https://github.com/user-attachments/assets/159ddb03-376a-4d4a-bfd4-77388d1ab821)

**Comment by zoifar** [comment]
edge case 1:
<img width="869" alt="image" src="https://github.com/user-attachments/assets/2f03cd38-293c-47fa-87f8-828e1ce46c42">
edge case 2:
<img width="842" alt="image" src="https://github.com/user-attachments/assets/8b9c316c-1221-4605-b1d5-f505438245fc">

**Comment by niccokunzmann** [comment]
This is confusing me at the moment... I wonder what should be expected. I found the edge cases but the calendars handle them differently. What are your thoughts on this? Obviously these edge cases have different results. 

This is the difference between the edge cases:

```
EXDATE;VALUE=DATE:20240715
```

**Comment by zoifar** [comment]
The way atlassian confluence calendar handles these cases at least looks logical. 
My guess is that other calendars somehow make similar changes differently and thus avoid this cases altogether. 
I think the best way forward is to handle the edge cases the way atlassian confluence calendar does. At lease until some other examples popes up sourced from another calendar app.

**Comment by niccokunzmann** [comment]
This is fixed in v3.1.1.

**Comment by niccokunzmann** [comment]
Funding is appreciated if you work for a company. Thanks for the well-written issues!
